openedx/edx-enterprise
# fix: customer sorting error

Add support for multiple user query such as a user's email or username and add support for ordering criteria.

## Testing
1. Add multiple learners and admins to an enterprise customer in django admin
2. Verify that endpoint http://localhost:18000/enterprise/api/v1/enterprise-customer-support/<enterprise_customer_uuid> returns admin users first followed learners. a user is an admin if the field `is_admin` is set to true and `enterprise_admin` is in field `role_assignments`. see example below
<img width="1250" alt="Screenshot 2024-09-03 at 12 11 45 AM" src="https://github.com/user-attachments/assets/a9b1d7da-952a-4af2-82dd-0b73e049b664">

3. Verify that endpoint http://localhost:18000/enterprise/api/v1/enterprise-customer-support/<enterprise_customer_uuid> ?user_query=<user_query> returns filtered user. see example below.
<img width="1230" alt="Screenshot 2024-09-03 at 12 12 31 AM" src="https://github.com/user-attachments/assets/45ca5922-7172-48b2-b86b-bd33ebc588f1">

5. Verify that endpoint http://localhost:18000/enterprise/api/v1/enterprise-customer-support/<enterprise_customer_uuid> ?ordering=learner returns learners first. see example below.
<img width="1230" alt="Screenshot 2024-09-03 at 12 12 31 AM" src="https://github.com/user-attachments/assets/3faa13a7-41cf-4e31-aaf5-1dd0a0603a12">


**Merge checklist:**
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

**Post merge:**
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - Trigger the '[Upgrade one Python dependency](https://github.com/openedx/edx-platform/actions/workflows/upgrade-one-python-dependency.yml)' action against master in edx-platform with new version number to generate version bump PR
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.
